lib/HLSL/DxilValidation.cpp

```diff
@@ -2345,7 +2345,83 @@ static void ValidateDxilOperationCallInProfile(CallInst *CI,
           CI, ValidationRule::SmOpcodeInInvalidFunction,
           {"CreateHandle", "Shader model 6.5 and below"});
     }
+  } break;
+    
+  // this intrinsic is generated when the SV_DispatchThreadID system value is used
+  case DXIL::OpCode::ThreadId: {  
+    // get the node launch type
+    DxilModule &DM = ValCtx.DxilMod;
+    if (!DM.HasDxilEntryProps(F)) {
```

Comment:
First, we need to check whether `shaderKind == DXIL::ShaderKind::Node`, since these intrinsics are used for other shader types as well.

Next, `DXIL::NodeLaunchType` should also be gathered at the top of this function when appropriate, since it will be useful for multiple verifications.  Also, at some point, we will also need to update the validator to handle noinline functions potentially called from entry points (a current gap), where this can be inherited through the call graph, rather than looked up directly on this function.